client/src/api.ts

```diff
@@ -25,35 +46,144 @@ export const createElection = async (data: {
       electionId: `uuid-${Date.now()}`, // This will be a UUID
     }
   }
-  return await fetchJSON('/elections', {
+  return await fetchJSON(`/api/mailelection`, {
     method: 'post',
     body: JSON.stringify(data),
   })
 }
 
-export const getElections = async () => {
+export const getElections = async (): Promise<{
+  mailElections: MailElection[]
+}> => {
   if (useFakeAPI) {
     await sleep()
-    return {
-      elections: [
-        {
-          electionId: 'asdf',
-        },
-      ],
-    }
+    return { mailElections: [mailElection] }
   }
-  return await fetchJSON('/elections')
+  return await fetchJSON(`/api/mailelection`)
 }
 
 export const getElection = async (
   electionId: string
-): Promise<ElectionsListItem> => {
+): Promise<MailElection> => {
   if (useFakeAPI) {
-    return {
-      id: electionId,
-      name: 'foo',
-      createdAt: new Date().toISOString(),
-    }
+    return mailElection
   }
-  return fetchJSON(`/elections/${electionId}`)
+  return fetchJSON(`/api/mailelection/${electionId}`)
 }
+
+export const approveElection = async (data: {
+  electionId: string
+  userId: string
+}) =>
+  await fetchJSON(`/api/mailelection/${data.electionId}/approve`, {
+    method: 'post',
+    body: JSON.stringify(data),
+  })
+
+export const putElectionDefinition = async ({
+  electionId,
+  election,
+}: {
+  electionId: string
+  election: Election
+}) =>
+  await fetchJSON(`/api/mailelection/${electionId}/definition`, {
+    method: 'put',
+    body: JSON.stringify(election),
+  })
+
+export const getElectionDefinition = async ({
+  electionId,
+}: {
+  electionId: string
+}): Promise<Election> =>
+  await fetchJSON(`/api/mailelection/${electionId}/definition`)
+
+export const putBallotTemplate = async ({
+  ballotStyleId,
+  electionId,
+  file,
+  precinctId,
+}: {
+  ballotStyleId: string
+  electionId: string
+  file: Buffer
+  precinctId: string
+}) =>
+  await fetchJSON(
+    `/api/mailelection/${electionId}/ballot-style/${ballotStyleId}/precinct/${precinctId}/template`,
+    {
+      method: 'put',
+      body: file,
+    }
+  )
+
+export const getBallotTemplate = async ({
+  ballotStyleId,
+  electionId,
+  precinctId,
+}: {
+  ballotStyleId: string
+  electionId: string
+  precinctId: string
+}): Promise<BallotTemplate> =>
+  await fetchJSON(
+    `/api/mailelection/${electionId}/ballot-style/${ballotStyleId}/precinct/${precinctId}/template`
+  )
+
+export const putMailingList = async ({
+  electionId,
+  voterMailingListFile,
+}: {
+  electionId: string
+  voterMailingListFile: Buffer
+}) =>
+  await fetchJSON(`/api/mailelection/${electionId}/mailinglist`, {
+    method: 'put',
+    body: voterMailingListFile,
+  })
```

Comment:
I'm okay with assuming we just replace the mailing list file every time for now. Later, we may need to add ability to add incremental mailing list files. But again, let's start small.

Reply:
Why even store the file at all then? Let's talk through this UX.

Reply:
It's good to have an audit trail of all the files that were uploaded in case they wonder how something happened and want to later download all the files.